src/main/java/co/novu/api/topics/TopicHandler.java

```diff
@@ -10,63 +10,73 @@
         if (request.getPageSize() != null) params.put("pageSize", request.getPageSize());
         if (request.getKey() != null) params.put("key", request.getKey());
-        if (params.isEmpty()) {
-            return restHandler.handleGet(FilterTopicsResponse.class, novuConfig, ENDPOINT);
-        }
-        return restHandler.handleGet(FilterTopicsResponse.class, novuConfig, ENDPOINT, params);
+        Response<FilterTopicsResponse> response = topicApi.filterTopics(params).execute();
+        return restHandler.extractResponse(response);
     }
 
-    public SubscriberAdditionResponse addSubscriberToTopic(SubscriberAdditionRequest request, String topicKey) {
-        return restHandler.handlePost(request, SubscriberAdditionResponse.class, novuConfig, ENDPOINT + "/" + topicKey + "/subscribers");
+    public SubscriberAdditionResponse addSubscriberToTopic(SubscriberAdditionRequest request, String topicKey) throws IOException, NovuNetworkException {
+        Response<SubscriberAdditionResponse> response = topicApi.addSubscriberToTopic(topicKey, request).execute();
+        return restHandler.extractResponse(response);
     }
 
-    public CheckTopicSubscriberResponse checkTopicSubscriber(String topicKey, String externalSubscriberId) {
-        return restHandler.handleGet(CheckTopicSubscriberResponse.class, novuConfig,ENDPOINT + "/" + topicKey +"/subscribers/" + externalSubscriberId);
+    public CheckTopicSubscriberResponse checkTopicSubscriber(String topicKey, String externalSubscriberId) throws IOException, NovuNetworkException {
+        Response<CheckTopicSubscriberResponse> response = topicApi.checkTopicSubscriber(topicKey, externalSubscriberId).execute();
+        return restHandler.extractResponse(response);
     }
 
-    public SubscriberRemovalResponse removeSubscriberFromTopic(SubscriberAdditionRequest request, String topicKey) {
-        boolean isSuccess = restHandler.handlePostForVoid(request, novuConfig, ENDPOINT + "/" + topicKey + "/subscribers/removal");
+    public SubscriberRemovalResponse removeSubscriberFromTopic(SubscriberAdditionRequest request, String topicKey) throws IOException {
+        Response<SubscriberRemovalResponse> response = topicApi.removeSubscriberFromTopic(topicKey, request).execute();
+        boolean isSuccess = response.isSuccessful();
         if (isSuccess) {
             return new SubscriberRemovalResponse();
         }
         return null;
     }
 
-    public DeleteTopicResponse deleteTopic(String topicKey) {
-        boolean isSuccess = restHandler.handleDeleteForVoid(novuConfig, ENDPOINT + "/" + topicKey);
+    public DeleteTopicResponse deleteTopic(String topicKey) throws IOException {
+        Response<DeleteTopicResponse> response = topicApi.deleteTopic(topicKey).execute();
+        boolean isSuccess = response.isSuccessful();
         if (isSuccess) {
             return new DeleteTopicResponse();
         }
         return null;
     }
```

Comment:
You don't need to check if it's successful here, there is a second **_extractResponse()_** function in RestHandler class. You should use that instead